homeassistant/components/sensor/worldtidesinfo.py

```diff
@@ -0,0 +1,135 @@
+"""
+Support for retrieving tide levels from worldtides.info.
+
+configuration.yaml
+
+sensor:
+  - platform: worldtidesinfo
+    api_key: "YOUR API KEY"
+    latitude: "LATITUDE OF BEACH"
+    longitude: "LONGITUDE OF BEACH"
+"""
+import logging
+import time
+from datetime import timedelta
+
+import homeassistant.helpers.config_validation as cv
+import requests
+import voluptuous as vol
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (CONF_API_KEY, CONF_LATITUDE, CONF_LONGITUDE)
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+_LOGGER = logging.getLogger(__name__)
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=3600)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_API_KEY): cv.string,
+    vol.Required(CONF_LATITUDE): cv.string,
+    vol.Required(CONF_LONGITUDE): cv.string,
+})
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the WorldTidesInfo sensor."""
+    data = WorldTidesInfoData(hass)
+    name = "WorldTidesInfo"
+
+    global _RESOURCE
+    global LAT
+    global LON
+    global KEY
+
+    LAT = config.get(CONF_LATITUDE)
+    LON = config.get(CONF_LONGITUDE)
+    KEY = config.get(CONF_API_KEY)
+
+    STARTTIME = int(time.time())
+    _RESOURCE = 'https://www.worldtides.info/api?extremes&length=86400&start=%s&lat=%s&lon=%s&key=%s' % (
+        STARTTIME, LAT, LON, KEY)
+
+    try:
+        data.update()
+    except RunTimeError:
+        _LOGGER.error("Unable to connect fetch WorldTidesInfo data: %s")
+        return False
+
+    add_devices([WorldTidesInfoSensor(data, name)])
```

Comment:
Set `True` as the [second argument](https://home-assistant.io/developers/code_review_platform/#5-entity).